sql/log.cc

```diff
@@ -5869,17 +5869,6 @@ THD::binlog_start_trans_and_stmt()
     this->binlog_set_stmt_begin();
     bool mstmt_mode= in_multi_stmt_transaction_mode();
 #ifdef WITH_WSREP
-    /*
-      With wsrep binlog emulation we can skip the rest because the
-      binlog cache will not be written into binlog. Note however that
-      because of this the hton callbacks will not get called to clean
-      up the cache, so this must be done explicitly when the transaction
-      terminates.
-    */
-    if (WSREP_EMULATE_BINLOG_NNULL(this))
```

Comment:
If you drop this optimization, then we should probably remove the patch at the beginning of ha_savepoint().